cmds/core/dhclient/dhclient.go

```diff
@@ -47,51 +48,63 @@ var (
 	v4Port = flag.Int("v4-port", dhcpv4.ServerPort, "DHCPv4 server port to send to")
 )
 
-func main() {
-	flag.Parse()
-	if len(flag.Args()) > 1 {
-		log.Fatalf("only one re")
+func run(args []string, timeout int, retry int, dryRun bool, verbose bool, vverbose bool, ipv4 bool, ipv6 bool, v6Port int, v6Server string, v4Port int) error {
+	if len(args) > 1 {
+		return fmt.Errorf("more than one interface specified")
 	}
 
-	if len(flag.Args()) > 0 {
-		ifName = flag.Args()[0]
+	if len(args) > 0 {
```

Comment:
is this check redundant with the `if len(args) > 1` check on line 52?   When the `if` statement on line 52 is true, the only way that we can reach this line is `len(args) == 1`

Reply:
The check is not redundant, since the we want to differ between `len(args)==0` and `len(args)==1`. I will change both checks into a switch case to make it more readable.

Reply:
oh, I totally missed that ifName was previously a package-level variable. Thanks for pointing this out!